source/grid/grid_tools_dof_handlers.cc

```diff
@@ -2063,8 +2063,8 @@ namespace GridTools
     Assert(0 <= direction && direction < space_dim,
            ExcIndexRange(direction, 0, space_dim));
 
-    Assert(pairs1.size() == pairs2.size(),
-           ExcMessage("Unmatched faces on periodic boundaries"));
+    // Assert(pairs1.size() == pairs2.size(),
+    //       ExcMessage("Unmatched faces on periodic boundaries"));
```

Comment:
How do we proceed with these? (We maybe said something, but I forgot again.)

Reply:
@kronbichler You have suggested to introduce here an if-statement, checking if we are dealing with a `p:f:t` or not. However, I do not have access here to the triangulation, do I?

Reply:
One could of course try a `dynamic_cast`, but the question is if we have a more basic way to figure these things out. In any case, we ought really keep the assertions for the existing variants.

Reply:
Done!